### PR TITLE
Replace dynamic invocation of EnumerateExpansionsAsync with direct calls

### LINKED ISSUE #7943: [PR Available] Remove IExpansionManager dynamic workaround
Introduced in https://github.com/dotnet/roslyn/pull/7935

I need to reference a type introduced in Update 1, but the build machines are still on RTM, so I'm working around it by calling the method I need dynamically. Once the build machines are on Update 1, I can remove the hack.

### PR DESCRIPTION
**Pull request by dpoeschl**
Fixes #7943

It is safe to mention IExpansionManager by name now that the build machines are on Visual Studio 2015 Update 1. This should not introduce any behavioral change (other than removing first-chance exceptions).

**Retargeting to `future`**

### CODE REVIEW

**Comment by dpoeschl** [comment]
Previous incarnations with various signoffs: https://github.com/dotnet/roslyn/pull/9213 https://github.com/dotnet/roslyn/pull/9009

**Comment by dpoeschl** [comment]
@dotnet/roslyn-ide

File: src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs [hunk 1]
```diff
@@ -123,64 +113,69 @@ private void PopulateSnippetCaches()
...
-                            TaskScheduler.Default).CompletesAsyncOperation(token);
+            var asyncExpansionManager = _expansionManager as IExpansionManager;
+            if (asyncExpansionManager != null)
+            {
+                // Call the asynchronous IExpansionManager API from a background thread
+                Task.Factory.StartNew(async () => await PopulateSnippetCacheAsync(asyncExpansionManager).ConfigureAwait(false),
+                                CancellationToken.None,
+                                TaskCreationOptions.None,
+                                TaskScheduler.Default).CompletesAsyncOperation(token);
+
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs]: Blank line.

**Comment by jasonmalinowski** [comment]
:+1:

**Comment by Pilchie** [comment]
:+1:

**Comment by Pilchie** [comment]
@dotnet-bot retest this please

@dpoeschl this is ready to go, once Jenkins is happy - right?

**Comment by dpoeschl** [comment]
Yep.

**Pull request merged**